imports/wasi_snapshot_preview1/fs.go

```diff
@@ -604,9 +682,17 @@ func fdReaddirFn(_ context.Context, mod api.Module, params []uint64) Errno {
 		return errno
 	}
 
-	// expect a cookie only if we are continuing a read.
 	if cookie == 0 && dir.CountRead > 0 {
-		return ErrnoInval // cookie is minimally one.
+		// This means that there was a previous call to the dir, but cookie is reset.
+		// This happens when the program calls rewinddir, for example:
+		// https://github.com/WebAssembly/wasi-libc/blob/659ff414560721b1660a19685110e484a081c3d4/libc-bottom-half/cloudlibc/src/libc/dirent/rewinddir.c#L10-L12
+		//
+		// Since we cannot unwind fs.ReadDirFile results, we re-open while keeping the same file descriptor.
+		f, err := fsc.ReOpenDir(fd)
```

Comment:
so I implemented ReOpenDi instead of keeping all the entries in f.ReadDir.Entries in order to retain the current logic, added unit test cover this in fs_test.go. But not sure which would be the better way to go @codefromthecrypt

Reply:
ping @codefromthecrypt I need your input here

Reply:
we should close the already opened one and replace it, right?

Reply:
yeah this does what I would expect it to. also I think there's a TODO somewhere of using the original open flags if we ever kept them (seems we do now), so more sweeps are possible when the dust settles.

Reply:
ok cool. glad this is also what you had in mind. actually I added a new fields so they we can open with the exact same flags https://github.com/tetratelabs/wazero/blob/dd9f9901cc996f37b43f4374fd366e6a74e6de0a/internal/sys/fs.go#L170-L172